1610520/remove_field.py

```diff
@@ -0,0 +1,214 @@
+    for i in [5, 5, 5, 5, 5]:
+        yield i
+
+
+@retry(
+    retryable_exceptions=[ClientError],
+    wait_time_generator=wait_times_access,
+    module_logger=logger,
+)
+def fix_data_in_s3(fields, bucket, s3_client, crash_data):
+    """Fix data in raw_crash file in S3."""
+    crashid = crash_data["crashid"]
+
+    path = (
+        "v2/raw_crash/%(entropy)s/%(date)s/%(crashid)s"
+        % {
+            "entropy": crashid[:3],
+            "date": date_from_ooid(crashid).strftime("%Y%m%d"),
+            "crashid": crashid
+        }
+    )
+    resp = s3_client.get_object(Bucket=bucket, Key=path)
+    raw_crash_as_string = resp["Body"].read()
+    data = json.loads(raw_crash_as_string)
+    should_save = False
+    for field in fields:
+        if field in data:
+            del data[field]
+            should_save = True
+
+    if should_save:
+        s3_client.upload_fileobj(
+            Fileobj=io.BytesIO(dict_to_str(data).encode("utf-8")), Bucket=bucket, Key=path
+        )
+        print("# s3: fixed raw crash")
+    else:
+        print("# s3: raw crash was fine")
+
+
+@retry(
+    retryable_exceptions=[ConnectionError],
+    wait_time_generator=wait_times_access,
+    module_logger=logger,
+)
+def fix_data_in_es(fields, es_conn, crash_data):
+    """Fix document in Elasticsearch."""
+    crashid = crash_data["crashid"]
+    index = crash_data["index"]
+
+    doc_type = es_conn.get_doctype()
+    with es_conn() as conn:
+        search = Search(using=conn, index=index, doc_type=doc_type)
+        search = search.filter("term", **{"processed_crash.uuid": crashid})
+        results = search.execute().to_dict()
+        result = results["hits"]["hits"][0]
+        index = result["_index"]
+        document_id = result["_id"]
+        document = result["_source"]
+
+        should_save = False
+        for field in fields:
+            if field in document["raw_crash"]:
+                should_save = True
+                del document["raw_crash"][field]
+
+        if should_save:
+            conn.index(
+                index=index, doc_type=doc_type, body=document, id=document_id
+            )
+            print("# es: fixed document")
+        else:
+            print("# es: document was fine")
+
+
+def fix_data(crashids, fields):
+    s3_context = get_s3_context()
+    bucket = s3_context.config.bucket_name
+    s3_client = s3_context.client
+
+    es_conn = get_es_conn()
+
+    for crashid in crashids:
+        print("# working on %s" % crashid)
+
+        # Fix the data in S3 and then Elasticsearch
+        try:
+            fix_data_in_s3(fields, bucket, s3_client, crashid)
+            fix_data_in_es(fields, es_conn, crashid)
+        except Exception:
+            # If this throws an exception, print it out and move on. Then we'll finish
+            # all the fixing for the first pass and can address the problematic crash
+            # reports in a second pass.
+            logger.exception("COUGH")
```

Comment:
I'd include the crash ID in the exception message, to avoid confusion if stdout and stderr print in a strange order. However, if you know more about the logger config, that may be unnecessary.

Reply:
I like that idea. I'll change that.